src/Lessons-api.Data/Repositories/TeacherRepository.cs

```diff
@@ -39,28 +40,31 @@ public async Task<TeacherEntity> AddTeacher(TeacherEntity teacherEntity)
             return teacherEntity;
         }
 
-        public async Task<TeacherEntity> UpdateTeacher(int id, TeacherEntity teacherEntity)
+        public async Task<TeacherEntity> UpdateTeacher(int id, UserEntity userEntity)
         {
-            var updatedTeacher = await _lessonsContext.Teachers.FindAsync(id);
+            var updatedTeacher = await _lessonsContext.Teachers.Include(t => t.User).SingleOrDefaultAsync(t => t.Id == id);
```

Comment:
Change this to FirstOrDefaultAsync

![image](https://user-images.githubusercontent.com/64907389/128723910-174c7913-77eb-4852-b8e5-78aa23e7ee7e.png)